sympy/concrete/summations.py

```diff
@@ -259,10 +259,49 @@ def _eval_summation(self, f, x):
 
     def is_convergent(self):
         """
-        Convergence tests are used for checking the convergence of
-        a series. There are various tests employed to check the convergence,
-        returns true if convergent and false if divergent and NotImplementedError
-        if can not be checked. Like divergence test, root test, integral test,
+        We divide the study of convergence of infinite sums and products in
+        two parts.
+
+        First Part:
+        One part is the question whether all the terms are well defined, i.e.,
+        they are finite in a sum and also non-zero in a product. Zero
+        is the analogy of (minus) infinity in products as:
+
+        .. math::
+
+            e^{-\infty} = 0
+
+        Second Part:
+        The second part is the question of convergence after infinities,
+        and zeros in products, have been omitted assuming that their number
+        is finite. This means that we only consider the tail of the sum or
+        product, starting from some point after which all terms are well
+        defined.
+
+        For example, in a sum of the form:
+
+        .. math::
+
+            \sum_{1 \leq i < \infty} \frac{1}{n^2 + an + b}
+
+        where a and b are numbers. The routine will return true even if there
+        are infinities in the term sequence (at most two). An analogous
+        product would be:
+
+        .. math::
+
+            \prod_{1 \leq i < \infty} e^{\frac{1}{n^2 + an + b}}
+
+        This is how convergence is interpreted. It is concerned with what
+        happens at the limit. Finding the bad terms is another independent
+        matter.
+
+        Note, It is responsibility of user to see that the sum or product
+        is well defined.
+
+        There are various tests employed to check the convergence, returns
```

Comment:
This, and the following lines could be reorganized.

... to check the convergence: divergence test, root test, ....

The function returns true if the sum is convergent, false if divergent, and ...